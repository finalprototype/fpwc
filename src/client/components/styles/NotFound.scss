@import '../../styles/colors.scss';

.title {
  font-size: calc(min(30vw, 144px));
  font-weight: 700;
  color: $gray-dark;
}

.subtitle {
  font-size: calc(min(4vw, 18px));
  color: $gray-light;
  padding-bottom: 40px;
}

.hero {
  margin: 20px 40px;
  max-width: 100%;
}
