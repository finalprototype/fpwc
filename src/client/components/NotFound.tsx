import React, { useEffect, useState } from "react";
import { useHistory } from "react-router-dom";

import PageContent from './ui/PageContent';
import FmvBackground from './ui/FmvBackground';
import Video from '../videos/glitch.mp4';
import SynthwaveFallback from '../images/fallback-synthwave.jpg';
import styles from "./styles/NotFound.scss";
import DeadLink from "../images/dead_link.png";

interface Props {
  seconds?: number;
}

const NotFound: React.FunctionComponent<Props> = (props: Props) => {
  const { seconds = 5 } = props;
  const [count, setCount] = useState(seconds);
  const history = useHistory();

  useEffect(() => {
    const timer = setTimeout(() => {
      if (count === 1) {
        history.push('/');
        return;
      }
      setCount(count - 1);
    }, 1000);
    return () => clearTimeout(timer);
  }, [count]);

  return (
    <>
      <FmvBackground
        videoSource={Video}
        imageFallback={SynthwaveFallback}
      />
      <PageContent flex centered>
        <div className={styles.title}>
          404
        </div>
        <div className={styles.subtitle}>
          (redirecting in {count})
        </div>
        <br />
        <img className={styles.hero} src={DeadLink} />
      </PageContent>
    </>
  );
};

export default NotFound;
