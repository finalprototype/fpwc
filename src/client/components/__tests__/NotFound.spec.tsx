import React from "react";
import { shallow } from "enzyme";

import NotFound from "../NotFound";

describe("NotFound", () => {
  it("renders", () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
