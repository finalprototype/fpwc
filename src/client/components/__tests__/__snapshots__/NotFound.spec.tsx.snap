// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders 1`] = `
<FmvBackground
  imageFallback="mock-file"
  videoSource="mock-file"
>
  <PageContent
    centered={true}
    flex={true}
    full={false}
  >
    <div
      className="title"
    >
      404
    </div>
    <div
      className="subtitle"
    >
      (redirecting in 
      5
      )
    </div>
    <br />
    <img
      alt=""
      className="hero"
      src="mock-file"
    />
  </PageContent>
</FmvBackground>
`;
